File: example/php/http_kline.php
<?php

$curl = curl_init();
//官网：https://qos.hk
//免费api key注册申请：https://qos.hk
curl_setopt_array($curl, array(
  CURLOPT_URL => 'https://api.qos.hk/kline?key=your-api-key',
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => '',
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => 'POST',
  CURLOPT_POSTFIELDS =>'{
    "kline_reqs":[
        {"c":"US:AAPL,TSLA","co":1,"a":0,"kt":1001},
        {"c":"SH:600519","co":1,"a":0,"kt":1001},
        {"c":"SZ:000001","co":1,"a":0,"kt":1001},
        {"c":"HK:700","co":1,"a":0,"kt":1001},
        {"c":"CF:BTCUSDT","co":2,"a":0,"kt":1001}
    ]
}',
  CURLOPT_HTTPHEADER => array(
    'Content-Type: application/json'
  ),
));

$response = curl_exec($curl);

curl_close($curl);
echo $response;

File: example/java/HttpKline.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class HttpKline {
    public static void main(String[] args) {
        try {
            HttpClient client = HttpClient.newHttpClient();
            // official website:https://qos.hk
            // Free API key registration application.https://qos.hk
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(new URI("https://api.qos.hk/kline?key=your-api-key"))
                    .header("Content-Type", "application/json")
                    .POST(HttpRequest.BodyPublishers.ofString("""
                    {
                        "kline_reqs":[
                            {"c":"US:AAPL,TSLA","co":1,"a":0,"kt":1001},
                            {"c":"SH:600519","co":1,"a":0,"kt":1001},
                            {"c":"SZ:000001","co":1,"a":0,"kt":1001},
                            {"c":"HK:700","co":1,"a":0,"kt":1001},
                            {"c":"CF:BTCUSDT","co":2,"a":0,"kt":1001}
                        ]
                    }
                    """))
                    .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            System.out.println(response.body());
        } catch (Exception e) {
            System.out.println("error:" + e.getMessage());
            e.printStackTrace();
        }
    }
}

